crates/ecs/src/lib.rs

```diff
@@ -417,22 +417,17 @@ impl Archetype {
             .and_then(borrow_component_vec_mut)
     }
 
-    /// Adds a component of type `ComponentType` to the specified `entity`.
+    /// Adds a component of type `ComponentType` to archetype.
     fn add_component<ComponentType: Debug + Send + Sync + 'static>(
         &mut self,
-        entity: Entity,
         component: ComponentType,
     ) -> ArchetypeResult<()> {
-        let entity_index = self
-            .entity_to_component_index
-            .get(&entity)
-            .ok_or(ArchetypeError::MissingEntityIndex(entity))?;
-
+        // new
         let mut component_vec = self.borrow_component_vec_mut::<ComponentType>().ok_or(
             ArchetypeError::CouldNotBorrowComponentVec(TypeId::of::<ComponentType>()),
         )?;
+        component_vec.push(Some(component));
```

Comment:
I had thoughts about doing this aswell however chose to not change it since it could break things in code I do not have full familliarity with. Will try to remove option and see if it's an easy fix or not

Reply:
👍 

If it turns out to be complicated, then we can just create a follow-up issue that @JacobBredin can work on since he's more familiar with that part of the code!

Reply:
Yup, that's one of the places that's quite horrible because of the nested options (which would be alleviated by removing one of the layers of `Option`). But I definitely understand why you didn't want to mess with that code lol

@JacobBredin then we'll make a follow-up issue where you can try to eliminate option / do the necessary refactors. That sound good?

Reply:
I'll make a follow up issue to fix this when the PR is merged 👍